pkg/controllers/streaming/processor_controller.go

```diff
@@ -422,6 +426,53 @@ func (r *ProcessorReconciler) constructDeploymentForProcessor(processor *streami
 		return nil, err
 	}
 
+	volumes := []corev1.Volume{}
+	volumeMounts := []corev1.VolumeMount{}
+	for _, stream := range streams {
+		if stream.Status.Binding.MetadataRef.Name != "" {
+			metadataVolumeName := fmt.Sprintf("processor-stream-%s-metadata", stream.Name)
```

Comment:
I don't think the name here needs to contain processor, does it? When/if this is extracted as a library function, the fact that the binding is from a processor, or a source or whatever doesn't matter. Once stream X is bound, it's the same data available however it is consumed.

Reply:
The name of the volume is only used to coordinate between the volume definition and the mount. I used the prefix to reduce the risk of colliding with an existing volume for the function container in the pod. The name is not visible inside the container at runtime.

Reply:
I think you missed my point: this piece of code should eventually be extracted as re-usable. The name could be totally opaque for what it's worth, and I think we agree on that. So all I'm saying is that inserting "processor" in it here confuses the reader on that fact.

Reply:
We're building this iteratively. When a library is abstracted out we can deal with library concerns. Until then, I'm not concerned with library concerns.